include/matx/core/make_sparse_tensor.h

```diff
@@ -38,18 +38,33 @@
 namespace matx {
 namespace experimental {
 
-// Helper method to create zero storage.
 template <typename T>
-__MATX_INLINE__ static auto
-makeDefaultNonOwningZeroStorage(index_t sz, matxMemorySpace_t space) {
-  T *ptr = nullptr;
-  assert(sz > 0);
-  matxAlloc((void **)&ptr, sz * sizeof(T), space, 0);
+__MATX_INLINE__ static void setZero(T *ptr, index_t sz,
+                                    matxMemorySpace_t space) {
   if (space == MATX_DEVICE_MEMORY || space == MATX_ASYNC_DEVICE_MEMORY) {
     cudaMemset(ptr, 0, sz * sizeof(T));
```

Comment:
There's a `cudaMemsetAsync` for the async case, but you would need a stream for that.

Reply:
Perhaps we should (eventually) require a stream in the factory methods too.....